cni/pkg/nodeagent/server_test.go

```diff
@@ -0,0 +1,352 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodeagent
+
+import (
+	"context"
+	"errors"
+	"net/netip"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
```

Comment:
Removed assert in favor of the internal assert lib - I don't agree at all about mocks leading to bad tests, especially when we're talking about things where we are directly interfacing with operating system I/O or external systems.

The mock Kube client we already have in Istio is very nice, for instance.

A component that is easy to mock under test is a good component, and a hard to mock component is a bad one - we had a lot of the latter in the prev. iteration, we have far fewer of those here.

If the choice is between building in-tree mock objects by hand versus using a mocking lib, the mocking lib is always a better choice, and offers much stronger assertion checks than DIY mocks do.

The existing CNI agent had rather poor code coverage which frequently led to subtle breakage, this replacement has significantly more around complex bits to preclude that. The better coverage we have gotten from mocks has been massively helpful in preventing breakage in our internal CNI versus the current upstream version.

If you want to suggest alternate code structures that doesn't use mocks/stubs at all (DIY or otherwise) that maintains coverage rates, certainly open to those.